### ISSUE TITLE
Selection cannot be seen

### ISSUE THREAD

**Issue by benogle** [madebysource/atom-color-highlighting#2]
You're using background color on the color elements. The selection regions are z-index'd below the text, so when the background color is totally opaque, the selection cannot be seen. 

I dont have a great solution right now. Maybe the color should be a bit transparent (`.9`?) so _something_ changes when it's selected. 

Love this package!

**Comment by petrbrzek** [comment]
I will try to listen for some event and add more opacity to background color. Dunno what event yet..

**Comment by benogle** [comment]
I admittedly havent tried this yet, but what about setting the opacity on the color you use to max of `.9`?

**Comment by MoOx** [comment]
I'm thinking about an other solution to fix this problem:
what about not highlithing the #hexa, but using a pseudo element instead ? No idea if it's possible with atom, but it can be great to just append a pseudo after (little square) that have the color (this point is also a good idea to eventually trigger a colorpicker) ?

**Comment by petrbrzek** [comment]
Update plugin and try it now. It's hiding background if selection occurs.

**Comment by benogle** [comment]
Good idea. It sorta works. I would probably take a little different approach, though. You basically need a function that takes one of those nodes and gives you back a screen `Range` object. Then you can just check to see if that range is in a selection.

Also that `selectedTextLen` global that's getting checked is super confusing. I'll see if I can write that function next week, and do a little pull req.

**Comment by benogle** [comment]
@MoOx I like the idea of a color picker. The little square might be an interesting experiment and work in place of the selection check.

**Comment by petrbrzek** [comment]
I've done some code refactoring in last two days and I've started to implement the color square idea. 
You can see the the current plugin appearance below. I think it looks fine. It is also prepared for the color picker implementation. Unfortunately, I found that I can't use native color picker `<input type="color">`. It just doesn't work. :( 
Do you know about some simple color picker implementation?
I like the Chrome dev tools color picker for example.

![plugin](http://f.cl.ly/items/00223p3t093N0q3M0z1V/Screen%20Shot%202014-03-04%20at%2016.28.25.png)

**Comment by MoOx** [comment]
You should take a look to this plugin https://github.com/thomaslindstrom/color-picker

**Comment by MoOx** [comment]
Just a thing: why the square is after the ; ? Because you can have multiple color (eg: linear-gradient definition)
